### PR TITLE
Added 4 examples

### PR DESCRIPTION
**Pull request by qvisten999**
Basics: Blinking, Fading, Button
Sensors: Photoresistor

### CODE REVIEW

**Review by jstroem — COMMENTED** [review]
Remove .DS_Store - Add a .gitignore 
Create a readme for each example with showing of the png's and how to run.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,27 @@
-# examples
-Collection of essential Toit applications
+# Examples
+
+Examples is a repository with Toit examples, which for instance are shown at [console.toit.io](https://console.toit.io/devices/DEVICE_ID/code).
```
Comment (floitsch) [inline README.md:3]:
```suggestion
A repository with Toit examples, which for instance are shown at [console.toit.io](https://console.toit.io/devices/DEVICE_ID/code).
```

File: basics/blinking/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Blinking example
+
+This example prints 'blink' and makes the LED blink with an 1 second frequency.
```
Comment (floitsch) [inline basics/blinking/README.md:3]: That's probably on me...
```suggestion
This example prints 'blink' and makes the LED blink with a 1 second frequency.
```

File: basics/blinking/blinking.toit [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+Blinking LED example code.
+
+This example prints 'blink' and makes the LED blink with an 1 second frequency.
```
Comment (floitsch) [inline basics/blinking/blinking.toit:4]:
```suggestion
This example prints 'blink' and makes the LED blink with a 1 second frequency.
```

File: basics/blinking/blinking.toit [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import gpio
+
+/// The GPIO pin the LED is connected to.
+LED ::= 19
+
+main:
+  led := gpio.Pin LED --output
+  while true:
+    print "blink"
+    LED.set 1
```
Comment (floitsch) [inline basics/blinking/blinking.toit:20]:
```suggestion
    led.set 1
```

File: basics/blinking/blinking.toit [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    sleep --ms=500
+    LED.set 0
```
Comment (floitsch) [inline basics/blinking/blinking.toit:22]:
```suggestion
    led.set 0
```

File: basics/button/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Button example
+
+This example prints 'Button clicked', when you press click the button.
```
Comment (floitsch) [inline basics/button/README.md:3]:
```suggestion
This example prints 'Button clicked', when you press the connected button.
```

File: basics/button/button.toit [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+/// The GPIO pin the button is connected to.
+BUTTON ::= 21
+
+main:
+  button ::= gpio.Pin BUTTON --input
+  while true:
+    button.wait_for 1
+    print "Button clicked"
+    sleep --ms=100  // Debounce
```
Comment (floitsch) [inline basics/button/button.toit:20]:
```suggestion
    sleep --ms=100  // Debounce.
```

File: basics/button/button.toit [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    // Wait for the button to be released.
+    button.wait_for 0
+    sleep --ms=100  // Debounce
```
Comment (floitsch) [inline basics/button/button.toit:23]:
```suggestion
    sleep --ms=100  // Debounce.
```

File: basics/blinking/blinking.toit [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/**
+Blinking LED example code.
+
+This example prints 'blink' and makes the LED blink with an 1 second frequency.
+
+How to wire the LED:
+    - Place a 220ohm resistor in between the anode of the LED and GPIO19 and connect the cathode to ground.
+    - The resistor will limit the current flowing through the LED and the ESP32, increasing their life.
```
Comment (floitsch) [inline basics/blinking/blinking.toit:8]:
```suggestion
    - The resistor limits the current that flows through the LED and the ESP32, increasing their life.
```

File: basics/fading/fading.toit [hunk 10]
```diff
@@ -0,0 +1,33 @@
+/**
+Fading LED example code.
+
+This example increases and decreases the lighting level of a LED over and over again.
+
+How to wire the LED: 
+  - Place a 220ohm resistor in between the anode of the LED and GPIO19 and connect the cathode to ground.
+  - The resistor will increase the life of your LED.
```
Comment (floitsch) [inline basics/fading/fading.toit:8]:
```suggestion
  - The resistor limits the current that flows through the LED and the ESP32, increasing their life.
```

File: basics/fading/fading.toit [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+*/
+
+import gpio.pwm as gpio
+import gpio
+
+/// Defining the GPIO pin for the LED.
```
Comment (floitsch) [inline basics/fading/fading.toit:14]:
```suggestion
/// The GPIO pin for the LED.
```

File: basics/fading/fading.toit [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+LED ::= 19
+
+/// The configured range of the PWM.
+DUTY_MIN ::= 0.0
+DUTY_MAX ::= 1.0
+CHANGE_RATE ::= 0.01  // The amount of change every 5ms.
+
+main:
+  step := CHANGE_RATE
+  duty := DUTY_MIN
```
Comment (floitsch) [inline basics/fading/fading.toit:24]: I would move those two just before the while loop.
They are modified in the loop, so moving them closer there makes sense to me.
I would even have an empty line between the non-changing and the changing local.

File: basics/fading/fading.toit [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  led_pwm ::= gpio.Pwm --frequency=100
```
Comment (floitsch) [inline basics/fading/fading.toit:25]: opinions differ, but I generally don't use final local vars. (But again: many disagree).
```suggestion
  led_pwm := gpio.Pwm --frequency=100
```

File: basics/fading/fading.toit [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  led := gpio.Pin LED
+  fading_led := led_pwm.start led
+  while true:
+    fading_led.set_duty_factor(duty)
+    duty += step
+    if not DUTY_MIN <= duty <= DUTY_MAX:
+      step = -step
```
Comment (floitsch) [inline basics/fading/fading.toit:32]: My mistake. I forgot to bring it back into duty range.
```suggestion
      step = -step
      duty += step
```

This means that we stay at the min and max for 2 cycles. I think that's ok.

File: sensors/photoresistor/README.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
+# Photoresistor example
+
+This example prints whether the lighting level is above the threshold.
```
Comment (floitsch) [inline sensors/photoresistor/README.md:3]:
```suggestion
This example prints whether the lighting level is above a given threshold.
```

File: sensors/photoresistor/photoresistor.toit [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+import gpio.adc
+
+/// The PIN for the photo resistor.
+PHOTO_RESISTOR_PIN ::= 32
+
+/// Lighting threshold. Change the threshold to fit your lighting conditions.
+THRESHOLD ::= 2.2
+
+main:
+  sensor := adc.Adc (gpio.Pin PHOTO_RESISTOR_PIN)
```
Comment (floitsch) [inline sensors/photoresistor/photoresistor.toit:21]: We could also do:
```suggestion
  pin := gpio.Pin PHOTO_RESISTOR_PIN
  sensor := adc.Adc pin
```

Up to you what you prefer.

**Review by floitsch — APPROVED** [review]
LGTM!

File: basics/fading/fading.toit (hunk 12 above)
Comment (qvisten999) [reply basics/fading/fading.toit:24]: Fixed

**Pull request merged**